beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/handler/UnreferencedHandler.java

```diff
@@ -0,0 +1,46 @@
+package com.expediagroup.beekeeper.cleanup.handler;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.expediagroup.beekeeper.cleanup.path.PathCleaner;
+import com.expediagroup.beekeeper.cleanup.path.aws.S3PathCleaner;
+import com.expediagroup.beekeeper.core.model.EntityHousekeepingPath;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+
+@Component
+public class UnreferencedHandler extends GenericHandler {
+
+  private static final LifecycleEventType EVENT_TYPE = LifecycleEventType.UNREFERENCED;
+  private S3PathCleaner s3PathCleaner;
+
+  @Autowired
+  public UnreferencedHandler(S3PathCleaner s3PathCleaner) {
+    super();
+    this.s3PathCleaner = s3PathCleaner;
+  }
+
+  @VisibleForTesting
+  public void setS3PathCleaner(S3PathCleaner cleaner) {
```

Comment:
Do you really need it for testing if you can pass this in to the constructor? If you do need it could it be made `protected` or default? instead of `public`?

Reply:
Dug into this a bit more - it looks like the underlying issue was my IDE overriding PathCleaner to final which was breaking mockito's setup for the pathcleaner on some tests. Moved this to be just a generic protected setter which fixes this issue.